app/src/main/java/com/example/tasky/feature_agenda/data/worker/EventWorker.kt

```diff
@@ -0,0 +1,203 @@
+
+            val event = eventAdapter.fromJson(jsonEvent) ?: return@withContext Result.failure()
+            val deletedPhotos = jsonPhotoList?.let {
+                photoAdapter.fromJson(jsonPhotoList)
+            } ?: emptyList()
+
+             when(action) {
+                "create" -> {
+                    val result = createEvent(event)
+                    getResult(result)
+                }
+
+                "get" -> {
+                    val result = getEvent(eventId ?: return@withContext Result.failure())
+                    getResult(result)
+                }
+
+                "update" -> {
+                    val result = updateEvent(event, deletedPhotos)
+                    getResult(result)
+                }
+
+                "delete" -> {
+                    val result = deleteEventAndAttendees(eventId ?: return@withContext Result.failure())
+                    getResult(result)
+                }
+
+                else -> Result.failure()
+            }
+        }
+    }
+
+    private fun <T> getResult(result: Resource<T>): Result {
+        return when(result) {
+            is Resource.Error -> {
+                when(result.message) {
+                    "Invalid Response" ->  Result.failure(
+                        workDataOf("error" to result.message)
+                    )
+                    "Couldn't reach server" -> Result.retry()
+
+                    else -> Result.failure(
+                        workDataOf("error" to result.message)
+                    )
+                }
+            }
+            is Resource.Success -> {
+                Result.success(
+                    workDataOf("data" to result.data)
+                )
+            }
+            else -> Result.failure()
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private suspend fun createEvent(event: AgendaItem.Event): Resource<Unit> {
+
+        val photoList = event.photos.map { photo ->
+            val file = File(photo.url)
+            val requestFile: RequestBody = file.asRequestBody("image/*".toMediaTypeOrNull())
+            MultipartBody.Part.createFormData("photos", file.name, requestFile)
+        }
+
+        return try {
+            api.createEvent(
+                EventRequest(
+                    id = event.eventId,
+                    title = event.eventTitle,
+                    description = event.eventDescription ?: "",
+                    from = event.from.toUtcTimestamp(),
+                    to = event.to.toUtcTimestamp(),
+                    remindAt = event.remindAt.toUtcTimestamp(),
+                    attendeeIds = event.attendees.map { it.userId }
+                ),
+                photoList
+            )
+            Resource.Success()
+        } catch(e: HttpException) {
+            Resource.Error("Invalid Response")
+        } catch(e: IOException) {
+            Resource.Error("Couldn't reach server")
+        }
+    }
```

Comment:
The functionality you put inside this worker is more that of a repository. It's not so obvious IMO that you'll find logic about how to create, update, get and delete events inside a worker. Instead, I'd put all this in a repo and then inject the repo in the worker, so that it can call its functions